src/python/pants/backend/jvm/targets/jar_dependency.py

```diff
@@ -51,16 +111,34 @@ def __new__(cls, org, name, rev=None, force=False, ext=None, url=None, apidocs=N
       the dependency itself should be downloaded and placed on the classpath
     :param list excludes: Transitive dependencies of this jar to exclude.
     :type excludes: list of :class:`pants.backend.jvm.targets.exclude.Exclude`
+    :param string base_path: absolute base path that a relative file url is based from.
```

Comment:
> I think it's important to remove the base_path argument and make jar definitions consistent.

Note build file api for `jar` hasn't changed, the wrapper class hides the `base_path` logic. the wrapped api `JarDependency` does change with the new param. 

Is the wrapped api change your concern?

If we want to kill `base_path` from `JarDependency`, only issue is when we need relative path for fingerprinting, we would have to assume some relative base: we can use`buildroot`, this way it serves the same purpose, w/o introducing a new param.